tests/SlmLocaleTest/Strategy/UriPathStrategyTest.php

```diff
@@ -409,15 +409,13 @@ public function testAssembleWithDefaultNotMatching()
         $this->event->setLocale('en');
         $this->event->setUri($uri);
 
+        $this->strategy->getRouter()->setBaseUrl('/nl');
         $this->strategy->setOptions([
             'default' => 'fr',
         ]);
         $this->strategy->assemble($this->event);
 
-        $expected = '/en/foo/bar/baz';
-        $actual   = $this->event->getUri()->getPath();
-
-        $this->assertSame($expected, $actual);
+        $this->assertSame('/en/foo/bar/baz', $this->event->getUri()->getPath());
```

Comment:
strategies should not change base path. Should have been `/nl/en/foo/bar/baz` instead of `/en/foo/bar/baz` we can't modify base path as it is just folder name and not an application route.

Reply:
`nl` is the language, it has to be changed to `en`

Reply:
but also it is base path ` $this->strategy->getRouter()->setBaseUrl('/nl');` baseurl is often alias or folder name where the application is installed and changing it will result in not found response. @kokspflanze  what if you set ` $this->strategy->getRouter()->setBaseUrl('/my-app');` what url it will generate?

@basz am I right here?

Reply:
i will add a test for this (when im home)

Document_root `/my-app`
Request `/my-app/nl/...`

Reply:
added 2 tests for this, with default language to other language and without default language

Reply:
so as I understand baseurl is `my-app/public` and you can't change it. But in tests you changing it.
let's say if you would have app in `my-app/en` instead of `my-app/public` now you would redirect to `my-app/nl` and it does not exist. Does make sense? 

Do I still not understand something? please rename your public folder so mething like `nl` or someting and test

Reply:
added more tests, like `testAssembleWithBasePathWithMatchingLanguageName`

if you rename your public directory to `nl` and you use the `default` configuration, sure it would not work and mostly bug.
but it works without the `default` configuration, but i dont think that someone is doing this kind of stuff.

i also added the `testAssembleWithBasePathWithMatchingLanguageName` in master and i get `my-app/nl/en/nl/foo/bar/baz` instead of `/my-app/nl/en/foo/bar/baz`. the problem also exist with the `public` directory.

Reply:
Okay, I looked deeper and now understand why you append language and change base path. As I understand that because you testing `assemble` event but then maybe you can also add tests for `found` event? Because `found` event is used for redirect and also alters baseUrl.

Reply:
added tests for found